### DISCUSSION TITLE
User invitation flow

### DISCUSSION THREAD

**Discussion by KeithGillette** [accounts-js/accounts#1222] [Q&A] [answered]
Our application allows a workspace owner to invite others to join. With Meteor, we could call [Accounts.createUser](https://docs.meteor.com/api/passwords.html#Accounts-createUser) without specifying a password to create a user account for the invitee and then call [Accounts.sendEnrollmentEmail](https://docs.meteor.com/api/passwords.html#Accounts-sendEnrollmentEmail) to notify the person of the invitation and have them set their initial password. How can we accomplish this flow in AccountsJS given that [MongoServicePassword.createUser](https://www.accountsjs.com/docs/api/database-mongo-password/classes/mongoservicepassword/#createuser) requires a password?

**Comment by pradel** [comment, marked as answer]
You could provide a totally random password that is just not used to bypass the typescript error? And continue with the same flow as before, sending a mail to the user so he can set his password.

**Reply by KeithGillette** [reply]
Yes, this appears to work. In fact, we can `// @ts-ignore` and include no password for an equivalent to the Meteor flow, as the underlying `createPassword` logic doesn't strictly require a password, so the TypeScript typings may be incorrect.

**Reply by mrcleanandfresh** [reply]
I've noticed in the accounts-password source code that there are a couple of places:

1. [accounts-password.ts#createUser](https://github.com/accounts-js/accounts/blob/master/packages/password/src/accounts-password.ts#L626-L634) has a check if the password is there, indicating it's optional.
2. [accounts-password.ts#passwordAuthenticator](https://github.com/accounts-js/accounts/blob/master/packages/password/src/accounts-password.ts#L694-L700) if you attempt to log in a user without a password set, you get a `PasswordAuthenticatorErrors.NoPasswordSet` error message.

I think [`CreateUserServicePassword`](https://github.com/accounts-js/accounts/blob/master/packages/types/src/types/services/password/create-user.ts) just needs to have the password be optional.

**Reply by KeithGillette** [reply]
Thanks, @mrcleanandfresh. I had discovered the same and opened issue #1223 to request that the TypeScript typings be corrected.
